src/base/utils/environment_p.cpp

```diff
@@ -280,3 +280,44 @@ Utils::Environment::Impl::CustomProfile::applicationSettings(const QString &name
     const QString settingsFileName {QDir(configLocation()).absoluteFilePath(name + QLatin1String(".conf"))};
     return SettingsPtr(new QSettings(settingsFileName, QSettings::IniFormat));
 }
+
+QString Utils::Environment::Impl::NoConvertConverter::fromPortablePath(const QString &portablePath) const
+{
+    return portablePath;
+}
+
+QString Utils::Environment::Impl::NoConvertConverter::toPortablePath(const QString &path) const
+{
+    return path;
+}
+
+Utils::Environment::Impl::Converter::Converter(const QString &basePath)
+    : m_baseDir {basePath}
+{
+    m_baseDir.makeAbsolute();
+}
+
+QString Utils::Environment::Impl::Converter::toPortablePath(const QString &path) const
+{
+    if (path.isEmpty() || m_baseDir.path().isEmpty())
+        return path;
+
+#ifdef Q_OS_WIN
+    if (QDir::isAbsolutePath(path)) {
+        QChar driveLeter = path[0].toUpper();
+        QChar baseDriveLetter = m_baseDir.path()[0].toUpper();
+        bool onSameDrive = (driveLeter.category() == QChar::Letter_Uppercase) && (driveLeter == baseDriveLetter);
```

Comment:
There are four options added by this PR. Two of them turn on independent features of the portable mode: `--profile` and `--relative-fastresume`. There is one option for configurations (`--configuration) and there is a shortcut `--profile` which is equal to `----profile=$EXE_DIR/profile` and `--elative-fastresume`.

For the use-case noted by @glassez one needs just `--profile` option. Those who use a portable drive for qbt and downloads need both options.